app/src/main/java/com/kickstarter/services/KSApolloClient.kt

```diff
@@ -548,88 +549,74 @@ class KSApolloClient(val service: ApolloClient) : ApolloClientType {
-            .id(decodeRelayId(backingGr?.location()?.id())?: 0)
-            .build()
 
-    val reward = Reward.builder()
-            .minimum(rewardAmount?: 0.0)
-            .description(backingGr?.reward()?.description())
-            .estimatedDeliveryOn(DateTime(backingGr?.reward()?.estimatedDeliveryOn()))
-            .shippingSingleLocation(rewardSingleLocation)
-            .id(rewardId)
+    val location = backingGr?.location()?.fragments()?.location()
+    val locationId = decodeRelayId(location?.id())
+    val projectId = decodeRelayId(backingGr?.project()?.fragments()?.project()?.id()) ?: 0
+    val shippingAmount = backingGr?.shippingAmount()?.fragments()
+
+    val reward = backingGr?.reward()?.fragments()?.reward()?.let { reward ->
+        val rewardId = decodeRelayId(reward.id()) ?: 0
+        val rewardAmount = reward.amount().fragments().amount().amount()?.toDouble()
+        val rewardSingleLocation = Reward.SingleLocation.builder()
+                .localizedName(location?.displayableName())
+                .id(decodeRelayId(location?.id())?: 0)
+                .build()
+
+        return@let Reward.builder()
+                .minimum(rewardAmount?: 0.0)
+                .description(reward.description())
+                .estimatedDeliveryOn(DateTime(reward.estimatedDeliveryOn()))
+                .shippingSingleLocation(rewardSingleLocation)
+                .id(rewardId)
+                .build()
+    }
+
+    val backerData = backingGr?.backer()?.fragments()?.user()
+    val nameBacker = backerData?.name()
+    val backerId= decodeRelayId(backerData?.id()) ?: 0
+    val avatar = Avatar.builder()
+            .medium(backerData?.imageUrl())
             .build()
 
     val backer = User.builder()
-            .id(id)
-            .name(name)
+            .id(backerId)
+            .name(nameBacker)
             .avatar(avatar)
             .build()
 
     return Backing.builder()
-            .amount(backingGr?.amount()?.amount().toString().toDouble())
+            .amount(backingGr?.amount()?.fragments()?.amount()?.amount()?.toDouble() ?: 0.0)
+            .paymentSource(payment)
             .backerId(backerId)
-            .backerUrl(avatar.medium())
-            .backerName(name)
-            .reward(reward)
-            .id(backerId)
+            .backerUrl(backerData?.imageUrl())
+            .backerName(nameBacker)
             .backer(backer)
+            .id(id)
+            .reward(reward)
+            .rewardId(reward?.id())
             .locationId(locationId)
-            .locationName(backingGr?.location()?.displayableName())
+            .locationName(location?.displayableName())
             .pledgedAt(backingGr?.pledgedOn())
             .projectId(projectId)
             .sequence(backingGr?.sequence()?.toLong() ?: 0)
-            .shippingAmount(backingGr?.shippingAmount()?.amount().toString().toFloat())
+            .shippingAmount(shippingAmount?.amount()?.amount()?.toFloat() ?: 0f)
```

Comment:
Should this still be optional or should be force to 0?

Reply:
It's really never gonna 0 but retrieving it from Graph force it to be optional because of the unwrapping. I'll change the defaults as Justin suggested to -1, that would be clear that if we encounter that value something went wrong